File: examples/java-maven/src/main/java/airco/HighTemperatureCondition.java
package airco;

import org.jeasy.rules.api.Condition;
import org.jeasy.rules.api.FactsMap;

public class HighTemperatureCondition implements Condition<FactsMap> {

    static HighTemperatureCondition itIsHot() {
        return new HighTemperatureCondition();
    }

    @Override
    public boolean evaluate(FactsMap facts) {
        Integer temperature = facts.get("temperature");
        return temperature > 25;
    }

}

File: examples/kotlin-gradle/src/main/java/airco/HighTemperatureCondition.kt
package airco

import org.jeasy.rules.api.Condition
import org.jeasy.rules.api.FactsMap

class HighTemperatureCondition : Condition<FactsMap> {


    override fun evaluate(facts: FactsMap): Boolean {
        val temperature = facts.get("temperature") as Int?
        return temperature!! > 25
    }

    companion object {

        internal fun itIsHot(): HighTemperatureCondition {
            return HighTemperatureCondition()
        }
    }

}
